shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -937,4 +942,105 @@ flutter::PointerDataDispatcherMaker PlatformView::GetDispatcherMaker() {
   };
 }
 
+// Ad hoc pretty printer.
+static std::ostream& operator<<(
+    std::ostream& os,
+    const fuchsia::ui::input::PointerEventPhase& phase) {
+  switch (phase) {
+    case fuchsia::ui::input::PointerEventPhase::ADD:
+      return os << "ADD";
+    case fuchsia::ui::input::PointerEventPhase::HOVER:
+      return os << "HOVER";
+    case fuchsia::ui::input::PointerEventPhase::DOWN:
+      return os << "DOWN";
+    case fuchsia::ui::input::PointerEventPhase::MOVE:
+      return os << "MOVE";
+    case fuchsia::ui::input::PointerEventPhase::UP:
+      return os << "UP";
+    case fuchsia::ui::input::PointerEventPhase::REMOVE:
+      return os << "REMOVE";
+    case fuchsia::ui::input::PointerEventPhase::CANCEL:
+      return os << "CANCEL";
+    default:
+      return os << "<unknown>";
+  }
+}
+
+// Ad hoc pretty printer.
+static std::ostream& operator<<(
+    std::ostream& os,
+    const fuchsia::ui::input::PointerEventType& type) {
+  switch (type) {
+    case fuchsia::ui::input::PointerEventType::TOUCH:
+      return os << "TOUCH";
+    case fuchsia::ui::input::PointerEventType::MOUSE:
+      return os << "MOUSE";
+    default:
+      return os << "<unexpected|unknown>";
+  }
+}
+
+// Ad hoc pretty printer.
+static std::ostream& operator<<(
+    std::ostream& os,
+    const fuchsia::ui::input::KeyboardEventPhase& phase) {
+  switch (phase) {
+    case fuchsia::ui::input::KeyboardEventPhase::PRESSED:
+      return os << "PRESSED";
+    case fuchsia::ui::input::KeyboardEventPhase::RELEASED:
+      return os << "RELEASED";
+    case fuchsia::ui::input::KeyboardEventPhase::CANCELLED:
+      return os << "CANCELLED";
+    case fuchsia::ui::input::KeyboardEventPhase::REPEAT:
+      return os << "REPEAT";
+    default:
+      return os << "<unknown>";
+  }
+}
+
+// Ad hoc pretty printer.
+static std::ostream& operator<<(std::ostream& os,
+                                const fuchsia::ui::input::InputEvent& event) {
+  os << "fuchsia.ui.input.InputEvent.";
+  switch (event.Which()) {
+    case fuchsia::ui::input::InputEvent::Tag::kPointer: {
+      const auto& ptr = event.pointer();
+      os << "PointerEvent: time=" << ptr.event_time
+         << ", device_id=" << ptr.device_id << ", pointer_id=" << ptr.pointer_id
+         << ", type=" << ptr.type << ", phase=" << ptr.phase << ", x=" << ptr.x
+         << ", y=" << ptr.y << ", trace_id="
+         << PointerTraceHACK(ptr.radius_major, ptr.radius_minor)
+         << ", buttons=" << ptr.buttons;
+      break;
+    }
+    case fuchsia::ui::input::InputEvent::Tag::kFocus: {
+      const auto& fcs = event.focus();
+      os << "FocusEvent: time=" << fcs.event_time
+         << ", focused=" << std::boolalpha << fcs.focused;
+      break;
+    }
+    case fuchsia::ui::input::InputEvent::Tag::kKeyboard: {
+      const auto& kbd = event.keyboard();
+      os << "KeyboardEvent: time=" << kbd.event_time
```

Comment:
This all seems like it could very easily leak personal information. Can you ensure that this is only enabled in debug builds and check with the privacy team to make sure logs like this are ok.